gui/static/js/components/addbot.js

```diff
@@ -99,28 +88,86 @@ export default class AddBot extends React.Component {
         });
     }
 
+
+    /*
+        Updates list of discoverable bots
+    */
     redoDiscoverList(data){
         let new_bots = [];
 
         for (let i = 0; i < data.length; i++) {
             //Trim the forward-slash
-            var ip_address = data[i].substring(1);
-            new_bots.push(ip_address)
+            var ip_address = data[i];
+            new_bots.push(ip_address);
         }
 
-        this.setState({discoveredBots: new_bots})
+        this.setState({discoveredBots: new_bots});
+    }
+
+    /*
+        Changes ip address in input box
+    */
+    changeIPAddress(ip) {
+        this.setState({ip: ip});
+    }
+
+    /*
+        Adds bot to basestation
+    */
+    addBot() {
+        var _this = this;
```

Comment:
Can we do `const _this = this;` because `var` is uncomfy especially since it's used practically everywhere